pkg/core/dao/dao.go

```diff
@@ -86,7 +90,9 @@ func (dao *Simple) GetBatch() *storage.MemBatch {
 // GetWrapped returns new DAO instance with another layer of wrapped
 // MemCachedStore around the current DAO Store.
 func (dao *Simple) GetWrapped() DAO {
-	return NewSimple(dao.Store)
+	d := NewSimple(dao.Store)
+	d.MPT = dao.MPT
```

Comment:
An interesting part is that this should work for `Cached` because of its storage item cache. Though it probably won't for the master branch.